Dev/154 155

it-hms opened the pull request:
closes #154 
closes #155

it-hms committed:
Formatting commit

google-java-format: version 1.23.0

it-hms committed:
Convert Boolean tag values to 1 or 0 for update

The Flexy UpdateTagForm request requires that boolean tag values are 1
or 0 and not true or false. This change updates the
EwonUpdateTagValuesRequest class to convert boolean tag values to 1 or
0.
Closes #155

it-hms committed:
Corrected DM tag quality string to int mapping

The tag quality values differ between DataMailbox and the instant
values from M2Web API. The former provides strings e.g., "good", "bad",
and the later provides a bit map consisting of multiple flags. The
connector maps DataMailbox responses to the instant value bitmaps.
This commit corrects the mapping.

it-hms commented:
@alexjhawk, These changes still need to be tested

alexjhawk commented on eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/requests/m2w/M2WebEwonUpdateTagValuesRequest.java:
```diff
@@ -79,6 +79,27 @@ public M2WebEwonUpdateTagValuesRequest(
+   * @param obj - Tag value object that should be converted to a string
+   * @return string representation of the tag value
+   */
+  public static String TagValToString(Object obj) {
```
```suggestion
  public static String tagValToString(Object obj) {
```

alexjhawk commented on eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/requests/m2w/M2WebEwonUpdateTagValuesRequest.java:
```diff
@@ -115,7 +136,7 @@ private static String getUpdateTagValuesRequestPath(
           .append(tagIndex)
           .append("=")
-          .append(tagNameValuePair.getValue().toString());
+          .append(TagValToString(tagNameValuePair.getValue()));
```
```suggestion
          .append(tagValToString(tagNameValuePair.getValue()));
```

alexjhawk reviewed (changes requested):
It looks good, aside from the two recommendations. We can talk about testing tomorrow.

it-hms committed:
Catch exceptions per tag, not per update

it-hms committed:
Add boolean tag value handling

For DataMailbox responses, values are parsed to java Objects. The values
 0 and 1 will be Doubles. The values True and False will be Booleans.
 Talk2M responses are parsed to Strings '1' and '0'. This addition will
 handle these possibilities.

it-hms committed:
Update the parsing of integer and long tag values

Tag values from DataMailbox responses are parsed into Objects, which
GSON makes Doubles. Tag values from Talk2M responses are parsed into
Strings. This change allows for both.

it-hms committed:
Added string for datamailbox tag type for DWORD

it-hms committed:
RELEASE: v2.0.1

benlbaum commented:
@it-hms @alexjhawk are we able to get v2.0.1 merged so that we can test the Issue #154 fix?

it-hms commented:
@alexjhawk ,
The changes you requested are completed. Could you please add a review?
